src/main/java/com/tstepnik/planner/service/TaskService.java

```diff
@@ -50,34 +50,24 @@ public Task addTask(Task task) {
 
     public Task updateTask(Task task, Long taskId) {
         User user = authService.getLoggedUser();
-        Optional<Task> checkedTask = findTaskById(taskId);
-        if (checkedTask.isEmpty()) {
-            return taskRepository.save(task);
-        } else if (checkedTask.get().getUserId().equals(user.getId())) {
-            Task updatedTask = checkedTask.get();
-            updatedTask.setImportance(task.getImportance());
-            updatedTask.setDescription(task.getDescription());
-            updatedTask.setDone(task.isDone());
-            return taskRepository.save(updatedTask);
+        Task checkedTask = taskRepository.getById(taskId);
+        if (!checkedTask.getUserId().equals(user.getId())) {
+            throw new AccessDeniedException("This task doesn't belongs to User");
```

Comment:
```suggestion
            throw new AccessDeniedException("This task doesn't belong to User");
```

Reply:
Also I would change the ex message to something more informative,like: `User with login xyz cannot edit task with id 11`, but if we don't want to expose this type of info (which is user login and task id). then this message should be fine.

Reply:
Ok I added specific comment to each exception.